x-pack/legacy/plugins/lens/public/indexpattern_plugin/dimension_panel/field_select.tsx

```diff
@@ -48,25 +48,41 @@ export function FieldSelect({
   );
 
   function isCompatibleWithCurrentOperation(col: BaseIndexPatternColumn) {
-    return incompatibleSelectedOperationType
-      ? col.operationType === incompatibleSelectedOperationType
-      : !selectedColumn || col.operationType === selectedColumn.operationType;
+    if (incompatibleSelectedOperationType) {
+      return col.operationType === incompatibleSelectedOperationType;
+    }
+    return !selectedColumn || col.operationType === selectedColumn.operationType;
   }
 
   const fieldOptions = [];
-  const fieldLessColumn = filteredColumns.find(column => !hasField(column));
-  if (fieldLessColumn) {
+  const bestFieldLessColumn = filteredColumns.find(
+    column => !hasField(column) && isCompatibleWithCurrentOperation(column)
+  );
+  if (bestFieldLessColumn) {
```

Comment:
It seems to me like this could be simplified:

```js
const fieldlessColumn =
  filteredColumns.find(column => !hasField(column) && isCompatibleWithCurrentOperation(column))
  ||  filteredColumns.find(column => !hasField(column));

if (fieldlessColumn) {
   // ... now, we only need a single if clause, and no else
}
```

```